lib/facter/facts/linux/cloud/provider.rb

```diff
@@ -6,10 +6,17 @@ module Cloud
       class Provider
         FACT_NAME = 'cloud.provider'
 
+        def initialize
+          @virtual = Facter::Util::Facts::VirtualDetector.new
+        end
+
         def call_the_resolver
-          az_metadata = Facter::Resolvers::Az.resolve(:metadata)
+          provider = case @virtual.platform
+                     when 'hyperv'
+                       'azure' unless Facter::Resolvers::Az.resolve(:metadata).empty?
+                     end
```

Comment:
[optional] Don't really need the `case when` for a single case, could be a simple `if`

Reply:
Was thinking this could be a bit future-proof for when we add more cloud providers